### PR TITLE
Change order of spans in captions

### PR DESCRIPTION
**Pull request by wilkus27**
We want to be able to decide in `config.scss` or `common-stuff.scss` file about order of all spans creating captions of figures and tables: 

image

At this moment we can change order only for `os-title-label`,`os-number` and `os-divider`.

### CODE REVIEW

File: recipes/mixins/_number.scss [hunk 1]
```diff
@@ -348,10 +348,16 @@
       @error "BUG: Invalid captionType '#{$captionType}'";
     }
   }
+
+  $partOne: nth($figTableConfig, 1);
+  $partTwo: nth($figTableConfig, 2);
+  $partThree: nth($figTableConfig, 3);
+  $partFour: nth($figTableConfig, 4);
```
Comment (tomjw64) [inline recipes/mixins/_number.scss]: Not a fan of having to manually access each index like this. Should probably use a loop for this.

File: recipes/books/_common-stuff.scss [hunk 2]
```diff
@@ -21,6 +21,13 @@ $Config_PartType_Figure_CaptionContent  : (os-title-label: "Figure ", os-number:
 $Config_PartType_Figure_CaptionContentAp: (os-title-label: "Figure ", os-number: counter(appendix, upper-alpha) counter(figure), os-divider: " "); //TODO: Make a counter for both chapter and appendix, and automate the switch between standard and upper alpha counting?
 $Config_PartType_Figure_CaptionContentPr: (os-title-label: "Figure ", os-number: counter(figure), os-divider: " ");
 
+// Here we can determine order of spans creating captions of figures and tabels.
+// bCaption - contains all spans defined in $Config_PartType_Table(Figure)_CaptionContent(Ap)(Pr),
+// bCaptionTitle - contains span with default title of figure or table
+// bCaptionDivider - contains span with blank space
+// bCaptionContent - contains span with default content of caption
+$Config_FigTable_CaptionContentOrder: (bCaption, bCaptionTitle, bCaptionDivider, bCaptionContent);
```
Comment (tomjw64) [inline recipes/books/_common-stuff.scss]: Interesting idea here. One drawback might be that it forces a config file user to know the names of the buckets already.

File: recipes/mixins/_order.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@function contentOrder($orderConfig, $bucketsNumber) {
+    $value: null;
+
+    @for $i from 1 to $bucketsNumber +1 {
```
Comment (tomjw64) [inline recipes/mixins/_order.scss]: This can be an `@each` loop. This would allow you to drop the `4` from the arguments of content order.

**Comment by wilkus27** [comment]
@helenemccarron @tomjw64 In 018eeee I modified function to use `@each` loop and to be useful also for document titles.

One check failed. I don't have account in Concourse to be able to check what is wrong.

**Comment by wilkus27** [comment]
**Decision:** this proposition will not be used as a solution for #1011. This PR could be closed.

**Pull request closed without merging**